### DISCUSSION TITLE
Does this extension support Format configuration?

### DISCUSSION THREAD

**Discussion by ken-okabe** [ionide/ionide-vscode-fsharp#1778] [Q&A] [answered]
So far, after googling a lot, all information I've got is

[nojaf](https://github.com/nojaf) commented [on Jan 26, 2020](https://github.com/fsprojects/fantomas/issues/650#issuecomment-578485704)

It may be possible to configure F# formatting by creating a file `fantomas-config.json` on the project root.
 
The JSON maybe looks like this:
```
{
  "IndentSpaceNum":4,
  "PageWidth":120,
  "SemicolonAtEndOfLine":false,
  "SpaceBeforeArgument":true ,
  "SpaceBeforeColon":false,
  "SpaceAfterComma":true ,
  "SpaceAfterSemicolon":true ,
  "IndentOnTryWith":false,
  "ReorderOpenDeclaration":false,
  "SpaceAroundDelimiter":true ,
  "KeepNewlineAfter":false,
  "MaxIfThenElseShortWidth":40,
  "StrictMode":false
}
```

The problem is, that I tried to set  `"IndentSpaceNum":4` for test, it does not work, and obviously, I can't find official documentation anywhere in this repo or fantomas.

The bottom line of my question is Does this extension support Format configuration?

**Comment by baronfel** [comment, marked as answer]
All formatting configuration is through editorconfig files, which Fantomas supports native. All formatting in Ionide is handled by Fantomas, but documenting the exact configuration can be difficult since the Fantomas integration is loosely coupled - we have no direct information about what config settings are available or used.

I believe that Fantomas is working on detailed docs for all settings right now.

**Reply by ken-okabe** [reply]
Thank you @baronfel 

Actually I myself have developed a couple of VSCode extensions, so I would categorize myself as rather knowledgeable of this area, but still can't figure out, so please give me more information.

By saying "through editorconfig files", did you mean the ones such as
 `fantomas.indent` in Extension Settings in [fantomas-fmt](https://marketplace.visualstudio.com/items?itemName=paolodellepiane.fantomas-fmt) ?

I gave it a try

`  "FSharp.fantomas.indent": 2` in `settings.json`

![image](https://user-images.githubusercontent.com/1316994/187014061-1ee6ff93-8edc-4bcb-bbbd-8dba7f36b432.png)

, but unrecognized.

Since especially mostly, F# starters will depend on this Ionide VS extension, and many would think some format configuration should be possible, I think it would be nicer at least to present a pointer where to tweak if it's supported.

Thanks a lot for your contributions.

**Reply by baronfel** [reply]
This is what I mean; https://fsprojects.github.io/fantomas/docs/end-users/Configuration.html

**Reply by ken-okabe** [reply]
Thanks, I tried everything with 

![image](https://user-images.githubusercontent.com/1316994/187015345-bcec8a2a-c165-4c5b-bb58-ee20f8875013.png)

with 
[EditorConfig for VS Code](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig) and Ionide have been installed and activated, but the format customization failed to work. The rest of type-annotation etc works as usual.

That's my experience so far, but as long as it's supported I will manage it.
Still I believe it would be much nicer for users the URL with some explanation should be add to the Document here.
 https://fsprojects.github.io/fantomas/docs/end-users/Configuration.html

Thanks.

**Reply by ken-okabe** [reply]
ok,

```
root = true

[*.fs]
```

is really needed here, and it works now. Thanks for your help.
